### PR TITLE
Radio interference mapper 2

### PR DESCRIPTION
**Pull request by oehm-smith**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -360,6 +360,12 @@
             <artifactId>spring-context-support</artifactId>
             <version>${spring.version}</version>
         </dependency>
+        <dependency>
```
Comment (heyagosper) [inline pom.xml]: doesn't this violate the DIFUI principle? (Didn't In Fact Use It)

File: src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/RadioInterferenceMapperTest.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        assertCommon(radioInterferencesDTO, radioInterferenceEntity, MappingDirection.FROM_ENTITY_TO_DTO);
+
+        // IRKJ_RadioInterference_WithToDate has an end/to date
+
+        String actualToDateFormatted = getDTODateFormattedAsString(radioInterferencesDTO2, ToFromDate.TO_DATE);
+        String expectedToDateFormatted = getEntityDateFormattedAsString(radioInterferenceEntity, ToFromDate.TO_DATE);
+        MatcherAssert.assertThat(actualToDateFormatted, Is.is(expectedToDateFormatted));
+    }
+
+    private void assertCommon(RadioInterferencesType radioInterferencesDTO, RadioInterference radioInterferenceEntity, MappingDirection mappingDirection) {
```
Comment (heyagosper) [inline src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/RadioInterferenceMapperTest.java]: posh tone

File: src/test/java/au/gov/ga/geodesy/support/utils/ToFromDate.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package au.gov.ga.geodesy.support.utils;
+
+/**
+ * Represents the direction of mapping, for example XML types to JPA entities
+ * or vice versa.
+ */
+public enum ToFromDate {
```
Comment (heyagosper) [inline src/test/java/au/gov/ga/geodesy/support/utils/ToFromDate.java]: There's already a MappingDirection enum if that's what you need

Comment (heyagosper) [inline src/test/java/au/gov/ga/geodesy/support/utils/ToFromDate.java]: Lazar's comment:

Comment (oehm-smith) [inline src/test/java/au/gov/ga/geodesy/support/utils/ToFromDate.java]: Ok I disagree.  

* I can't use MappingDirection as it doesn't "map" that well (excuse the pun) to dates - is `FROM_DTO_TO_ENTITY` TO or FROM?  You could argue it does but at the cost of Brain Meltdown.  And what for - just so we can remove an extra Enum from the TEST source!
* I removed one use of TO_FROM but the other one is used in this method:

    private String getDTODateFormattedAsString(RadioInterferencesType radioInterferencesDTO, ToFromDate toFromDate) {
        String dateFormattedString = null;

        if (toFromDate == ToFromDate.FROM_DATE) {
            dateFormattedString = ((TimePeriodType) radioInterferencesDTO.getValidTime().getAbstractTimePrimitive().getValue()).getBeginPosition().getValue().get(0);
        } else {
            dateFormattedString = ((TimePeriodType) radioInterferencesDTO.getValidTime().getAbstractTimePrimitive().getValue()).getEndPosition().getValue().get(0);
        }
        Instant intermediateDate = GMLDateUtils.stringToDateMultiParsers(dateFormattedString);
        String dateFormatted = GMLDateUtils.dateToString(intermediateDate, format);
        return dateFormatted;
    }

And thus I'd be required to put `((TimePeriodType) radioInterferencesDTO.getValidTime().getAbstractTimePrimitive().getValue()).getEndPosition().getValue().get(0);` etc or similar directly in the test, which violates why I have this method in the first place - to move the unwieldy code to make the test easier to read.  The other alternative is to have two methods - one a "TO", another for "FROM".  But is this a lesser cost than having a TEENSY_WEENY_TINY ENUM in the TEST source?

File: src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/RadioInterferenceMapperTest.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+import org.hamcrest.core.IsNull;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Tests the mapping of a GeodesyML humiditySensor element
```
Comment (heyagosper) [inline src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/RadioInterferenceMapperTest.java]: copy paste error

File: src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/RadioInterferenceMapperTest.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                .findFirst().get();
+
+        RadioInterferencesType radioInterferencesDTO = logItem.getRadioInterferencesSet().get(0).getRadioInterferences();
+
+        RadioInterference radioInterferenceEntity = mapper.to(radioInterferencesDTO);
+
+        assertCommon(radioInterferencesDTO, radioInterferenceEntity, MappingDirection.FROM_DTO_TO_ENTITY);
+
+        // IRKJ_RadioInterference_NoToDate has no end/to date
+        MatcherAssert.assertThat(((TimePeriodType) radioInterferencesDTO.getValidTime().getAbstractTimePrimitive().getValue()).getEndPosition().getValue().size(), Is.is(0));
```
Comment (heyagosper) [inline src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/RadioInterferenceMapperTest.java]: could we check the line length rule

**Comment by oehm-smith** [comment]
New code checked in.  Please review.

File: src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/RadioInterferenceMapperTest.java [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+        }
+    }
+
+    /**
+     * Use date intermediate to guarantee success (ie. no formatting errors)
+     *
+     * @param radioInterferencesDTO
+     * @return date as string extracted from the DTO object and format using the internal format being used
+     */
+    private String getDTODateFormattedAsString(RadioInterferencesType radioInterferencesDTO, ToFromDate toFromDate) {
```
Comment (heyagosper) [inline src/test/java/au/gov/ga/geodesy/support/mapper/orika/geodesyml/RadioInterferenceMapperTest.java]: why not make a separate class that contains the dates test, this looks like some code that would be useful for other tests as well and could be reused.

File: src/test/java/au/gov/ga/geodesy/support/utils/ToFromDate.java (hunk 3 above)
Comment (heyagosper) [inline src/test/java/au/gov/ga/geodesy/support/utils/ToFromDate.java]: like my earlier comment, if the dates testing was in a separate file then it's reusable and then you're not saving much or any lines of code by having an enum

**Comment by oehm-smith** [comment]
Sorry I had to remove a file out of a commit so I've pushed it all again.  All requested changes in there.

**Pull request merged**